### PR TITLE
Fix multimodal chatinterface api bug

### LINKED ISSUE #8961: Audio streaming model not show waveform graph
### Describe the bug

Audio non-streaming model is display the audio waveform graph, but not show it when using streaming;

e.g:
non-steraming is show the waveform graph
image

streaming mode:
image

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
import gradio as gr

Audio(streaming=True, autoplay=True)
```

### Screenshot

_No response_

### Logs

_No response_

### System Info

```shell
gradio 3.40.0
```

### Severity

I can work around it

### LINKED ISSUE #8981: MultimodalTextbox expects a dictionary with optional keys 'text' and 'files'. Received str
### Describe the bug

Even if i supplied a dictionary, it still says treats as str.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```python
from gradio_client import Client
dict = {
    "text":"hello there",
    "files":[],
}

# needs to supply {"text":"txt", "files": []} in message
result = client.predict(
                message=dict,
                (...)
                api_name="/chat"
)
```

### Screenshot

n/a

### Logs

```shell
gradio_client.exceptions.AppError: The upstream Gradio app has raised an exception: MultimodalTextbox expects a dictionary with optional keys 'text' and 'files'. Received str
```

### System Info

```shell
gradio_client-1.2.0
```

### Severity

Blocking usage of gradio

### PR DESCRIPTION
**Pull request by freddyaboulton**
## Description

Closes #8981

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/chat_interface.py [hunk 1]
```diff
@@ -697,65 +697,6 @@ async def _stream_fn(
                 history_with_input[-1] = response  # type: ignore
                 yield history_with_input
 
-    async def _api_submit_fn(
```
Comment (freddyaboulton) [inline gradio/chat_interface.py:700]: These are not used anywhere?

**Comment by abidlabs** [comment]
Btw which issue is this pr addressing @freddyaboulton? I think https://github.com/gradio-app/gradio/issues/8961 is something else

**Comment by freddyaboulton** [comment]
Whoops its https://github.com/gradio-app/gradio/issues/8981

**Comment by abidlabs** [comment]
LGTM, tested with:

```py
from gradio_client import Client
import gradio as gr

_, url, _ = gr.ChatInterface(lambda x,y:x["text"], multimodal=True).launch(show_error=True, inline=False)

client = Client(url)

dict = {
    "text":"hello there",
    "files":[],
}

result = client.predict(
    x=dict,
    api_name="/chat"
)
```

**Comment by abidlabs** [comment]
Will merge this in for the release!

**Pull request merged**